### PR TITLE
Add a real dependency on Abseil

### PR DESCRIPTION
**Pull request by mkruskal-google**
This just replaces std::unordered_set with absl::flat_hash_set in a single file, to prove that the Abseil dependency works.

### CODE REVIEW

**Review by acozzette — APPROVED** [review]

File: src/google/protobuf/compiler/cpp/helpers.cc [hunk 1]
```diff
@@ -41,9 +41,9 @@
 #include <map>
 #include <memory>
 #include <queue>
-#include <unordered_set>
 #include <vector>
 
+#include "absl/container/flat_hash_set.h"
```
Comment (fowles) [inline src/google/protobuf/compiler/cpp/helpers.cc:46]: feels weird to put it at this point in the include list, but sure

File: src/google/protobuf/compiler/cpp/helpers.cc [hunk 2]
```diff
@@ -176,15 +176,15 @@ static const char* const kKeywordList[] = {
...
+static absl::flat_hash_set<std::string>* MakeKeywordsMap() {
+  auto* result = new absl::flat_hash_set<std::string>();
   for (const auto keyword : kKeywordList) {
     result->emplace(keyword);
   }
   return result;
 }
 
-static std::unordered_set<std::string>& kKeywords = *MakeKeywordsMap();
+static absl::flat_hash_set<std::string>& kKeywords = *MakeKeywordsMap();
```
Comment (fowles) [inline src/google/protobuf/compiler/cpp/helpers.cc:187]: ugh, this is not safe to have at global scope

File: src/google/protobuf/compiler/cpp/helpers.cc [hunk 3]
```diff
@@ -1116,7 +1116,7 @@ bool IsAnyMessage(const Descriptor* descriptor, const Options& options) {
 }
 
 bool IsWellKnownMessage(const FileDescriptor* file) {
-  static const std::unordered_set<std::string> well_known_files{
+  static const absl::flat_hash_set<std::string> well_known_files{
```
Comment (fowles) [inline src/google/protobuf/compiler/cpp/helpers.cc:1119]: global with destructor is also not safe.  Can you file a bug for both of these (preexisting) issues?

**Review by fowles — APPROVED** [review]

File: src/google/protobuf/compiler/cpp/helpers.cc (hunk 1 above)
Comment (mkruskal-google) [reply src/google/protobuf/compiler/cpp/helpers.cc:46]: I was aiming for where it won't get reordered in google3.  "third_party/absl" < "third_party/protobuf" :)

File: src/google/protobuf/compiler/cpp/helpers.cc (hunk 2 above)
Comment (mkruskal-google) [reply src/google/protobuf/compiler/cpp/helpers.cc:187]: ew good point, I'll just fix these now

File: src/google/protobuf/compiler/cpp/helpers.cc (hunk 3 above)
Comment (mkruskal-google) [reply src/google/protobuf/compiler/cpp/helpers.cc]: Fixed :)

**Pull request merged**